### PR TITLE
implemented guard let, JSONDecoder, code keys, combined 2 fetching funcs

### PR DESCRIPTION
**Pull request by k-code00**
fetched eth price (crypto) and printed it in usd, eur, gbp (currency)

### CODE REVIEW

File: DeBlockWallet/Model.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//
+//  Model.swift
+//  DeBlockWallet
+//
+//  Created by Consultant on 19/09/2023.
+//
+
+import Foundation
+
+//struct Crypto: Decodable {
```
Comment (mattmaloof) [inline DeBlockWallet/Model.swift:10]: Should not have so much commented code.

File: DeBlockWallet/NetworkManager.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    
+    // API Key
+    let apiKey = "YourApiKeyToken"
+    
+    // URL to fetch currency
+    let ethPrice = "https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd,eur,gbp"
+    
+    // URL to fetch gas prices
+    let gasPrice = "https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=YourApiKeyToken"
+    //let gasPrice = "https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=\(apiKey)"
```
Comment (mattmaloof) [inline DeBlockWallet/NetworkManager.swift:20]: Extra commented code here.

File: DeBlockWallet/NetworkManager.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            do {
+                let decodedResponse = try JSONDecoder().decode(ApiResponse.self, from: data)
+                guard let fastGasPrice = Double(decodedResponse.result.FastGasPrice) else {
+                    print("Error converting FastGasPrice to Double")
+                    return
+                }
+                let transactionFee = 21000.0 * fastGasPrice / 100000000.0
+                print("Approximate Ethereum transaction fee for a standard transfer with FastGasPrice: \(transactionFee) ETH")
+            } catch {
+                print("Failed to decode JSON:", error.localizedDescription)
```
Comment (mattmaloof) [inline DeBlockWallet/NetworkManager.swift:69]: Use closure, need to return an error and need to use completion handler to communicate the data between classes.

File: DeBlockWallet/NetworkManager.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    
+    func fetchEthereumPrice() {
+        fetchDataFromURL(ethPrice) { data in
+            do {
+                let decodedResponse = try JSONDecoder().decode(Crypto.self, from: data)
+                print("Ethereum Prices:")
+                print("USD: \(decodedResponse.ethereum.usd)")
+                print("EUR: \(decodedResponse.ethereum.eur)")
+                print("GBP: \(decodedResponse.ethereum.gbp)")
+            } catch {
```
Comment (mattmaloof) [inline DeBlockWallet/NetworkManager.swift:52]: Return error, use completion handler / closure to communicate code between classes.

**Review by mattmaloof — CHANGES_REQUESTED** [review]

**Review by ChopperTheKing — COMMENTED** [review]
Looks good so far.

**Review by mattmaloof — APPROVED** [review]

**Pull request merged**